pkgs/stdenv/generic/setup.sh

```diff
@@ -174,6 +186,52 @@ addToSearchPath() {
     addToSearchPathWithCustomDelimiter ":" "$@"
 }
 
+# Prepend elements to variable "$1", which may come from an attr.
+#
+# This is useful in generic setup code, which must (for now) support
+# both derivations with and without __structuredAttrs true, so the
+# variable may be an array or a space-separated string.
+#
+# Expressions for individual packages should simply switch to array
+# syntax when they switch to setting __structuredAttrs = true.
+_prepend() {
+    local -n nameref="$1"; shift
+    if [ -n "$__structuredAttrs" ]; then
+        nameref=( "$@" ${nameref+"${nameref[@]}"} )
+    else
+        nameref="$* ${nameref-}"
+    fi
+}
+
+# Accumulate into `flagsArray` the flags from the named variables.
+#
+# If __structuredAttrs, the variables are all treated as arrays
+# and simply concatenated onto `flagsArray`.
+#
+# If not __structuredAttrs, then:
+#   * Each variable is treated as a string, and split on whitespace;
+#   * except variables whose names end in "Array", which are treated
+#     as arrays.
+_accumFlagsArray() {
+    local name
+    if [ -n "$__structuredAttrs" ]; then
+        for name in "$@"; do
+            eval 'flagsArray+=( ${'$name'+"${'$name'[@]}"} )'
```

Comment:
This too.

```suggestion
            local -n nameref="$name"
            flagsArray+=( ${nameref+"${nameref[@]}"} )
```

Reply:
btw shellcheck is giving me this warning https://www.shellcheck.net/wiki/SC2178 should it be ignored?

Reply:
Yeah, looks like that’s koalaman/shellcheck#2060.